app/forms/conversion/create_project_form.rb

```diff
@@ -41,31 +43,25 @@ def initialize(params = {})
   end
 
   def provisional_conversion_date=(hash)
-    @provisional_conversion_date = Date.new(year_for(hash), month_for(hash), day_for(hash))
+    @provisional_conversion_date = Date.new(value_at_position(hash, 1), value_at_position(hash, 2), value_at_position(hash, 3))
   rescue NoMethodError
     nil
   rescue TypeError, Date::Error
     @attributes_with_invalid_values << :provisional_conversion_date
   end
 
   def advisory_board_date=(hash)
-    @advisory_board_date = Date.new(year_for(hash), month_for(hash), day_for(hash))
+    @advisory_board_date = Date.new(value_at_position(hash, 1), value_at_position(hash, 2), value_at_position(hash, 3))
   rescue NoMethodError
     nil
-  rescue TypeError, Date::Error
+  rescue TypeError, Date::Error, NegativeValueError
     @attributes_with_invalid_values << :advisory_board_date
   end
 
-  private def day_for(value)
```

Comment:
Just interested in why you got rid of these?

Reply:
Just to reduce duplication - I'd have to throw the NegativeDateError from each one and it felt better to roll them into one method. Although it's probably less readable now?